Dockerfile

```diff
@@ -1,12 +1,56 @@
-FROM jpetazzo/dind
+FROM jpetazzo/dind AS builder
 
 RUN cd /opt && \
   curl -SL http://www.edgeworx.io/downloads/jdk/jdk-8u211-64.tar.gz \
   | tar -xzC /opt && \
   update-alternatives --install /usr/bin/java java /opt/jdk1.8.0_211/bin/java 1100
 
-COPY packaging/iofog-agent/etc /etc
-COPY packaging/iofog-agent/usr /usr
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends unzip && \
+    apt-get clean
+
+COPY . .
+
+# 1- Define a constant with the version of gradle you want to install
+ARG GRADLE_VERSION=5.4
+
+# 2- Define the URL where gradle can be downloaded from
+ARG GRADLE_BASE_URL=https://services.gradle.org/distributions
+
+# 3- Define the SHA key to validate the gradle download
+#    obtained from here https://gradle.org/release-checksums/
+ARG GRADLE_SHA=c8c17574245ecee9ed7fe4f6b593b696d1692d1adbfef425bef9b333e3a0e8de
+
+# 4- Create the directories, download gradle, validate the download, install it, remove downloaded file and set links
+RUN mkdir -p /usr/share/gradle /usr/share/gradle/ref \
+  && echo "Downlaoding gradle hash" \
+  && curl -fsSL -o /tmp/gradle.zip ${GRADLE_BASE_URL}/gradle-${GRADLE_VERSION}-bin.zip \
+  \
+  && echo "Checking download hash" \
+  && echo "${GRADLE_SHA}  /tmp/gradle.zip" | sha256sum -c - \
+  \
+  && echo "Unziping gradle" \
+  && unzip -d /usr/share/gradle /tmp/gradle.zip \
+   \
+  && echo "Cleaning and setting links" \
+  && rm -f /tmp/gradle.zip \
+  && ln -s /usr/share/gradle/gradle-${GRADLE_VERSION} /usr/bin/gradle
+
+# 5- Define environmental variables required by gradle
+ENV GRADLE_VERSION 5.4
+ENV GRADLE_HOME /usr/bin/gradle
+ENV GRADLE_USER_HOME /cache
+ENV PATH $PATH:$GRADLE_HOME/bin
+
+VOLUME $GRADLE_USER_HOME
+
+RUN gradle build copyJar
+
+FROM jpetazzo/dind
+
+COPY --from=builder packaging/iofog-agent/etc /etc
+COPY --from=builder packaging/iofog-agent/usr/bin /usr/bin
```

Comment:
Not needed since you are copying the whole /usr folder at line 53

Reply:
updated